js/src/locale.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * Simple module for localization of strings.
+ *
+ * How to use:
+ * 1. Create localization file with following JSON format:
+ * {
+ *   "en" :{
+ *     "how": "How do you want your egg today?",
+ *     "boiledEgg": "Boiled egg",
+ *     "softBoiledEgg": "Soft-boiled egg",
+ *     "choice": "How to choose the egg",
+ *     "buyQuestion": "I'd like to buy {0} eggs, {1} dollars each."
+ *   },
+ *
+ *   "it": {
+ *     "how": "Come vuoi il tuo uovo oggi?",
+ *     "boiledEgg": "Uovo sodo",
+ *     "softBoiledEgg": "Uovo alla coque",
+ *     "choice": "Come scegliere l'uovo",
+ *     "buyQuestion": "Mi piacerebbe comprare {0} uova, {1} dollari ciascuna."
+ *   }
+ * }
+ * 
+ * 2. Use Locale's module load method, specifying path to your localization file and set chosen language, e.g.:
+ *    > Locale.load('static/locale/locale.json');
+ *    > Locale.setLanguage('en');
+ * 
+ * 3. Use localization strings:
+ *    > console.log(Locale.tr.how);
+ *    How do you want your egg today?
+ *    > console.log(Locale.tr.boiledEgg);
+ *    Boiled egg
+ * 
+ * 4. String formatting
+ *    > console.log(Locale.tr.buyQuestion.format(10, 0.5));
+ *    I'd like to buy 10 eggs, 0.5 dollars each.
+ */
+
+export class Locale {
+    /**
+     * Loads translation object from external json file.
+     *  
+     * @param {String} path Path to resource.
+     * @return {Promise}
+     */
+    static async load(path) {
+        await fetch(path)
+        .then((resp) => resp.json())
```

Comment:
We can remove the then chain by: 

```
const response = await fetch(path)
const obj = await response.json();
this.loadFromObject(obj);
```

Reply:
That's true but this creates local variables unnecessarily. Another alternative would be await expressions, but they cannot be chained nicely and are even less readable.

Reply:
fair point, lets just leave it as is